src/js/components/form/FormGroupHeading.js

```diff
@@ -0,0 +1,56 @@
+import classNames from 'classnames';
+import React from 'react';
+
+import FormGroupHeadingContent from './FormGroupHeadingContent';
+
+function getChildren(children, required) {
+  if (!required) {
+    return children;
+  }
+
+  // We add the required field indicator after the first child.
+  return React.Children.toArray(children).reduce(
+    function (accumulator, child, index) {
+      accumulator.push(child);
+
+      if (index === 0) {
+        accumulator.push(
+          <FormGroupHeadingContent
+            className="text-danger"
+            key={`${index}-asterisk`}>
+            *
+          </FormGroupHeadingContent>
+        );
+      }
+
+      return accumulator;
+    },
+    []
+  );
+}
+
+function FormGroupHeading({className, children, required}) {
+  const classes = classNames('form-group-heading', className);
+
+  return (
+    <div className={classes}>
+      {getChildren(children, required)}
+    </div>
+  );
+}
+
+FormGroupHeading.defaultProps = {
+  required: false
+};
+
+FormGroupHeading.propTypes = {
+  children: React.PropTypes.node.isRequired,
+  className: React.PropTypes.oneOfType([
```

Comment:
yep, we definitely need a propType for that, let's create a small tech debt jira